### ISSUE TITLE
Crash: one of my repos fails to load

### ISSUE THREAD

**Issue by SlyEcho** [sourcegit-scm/sourcegit#636]
The UI is showing a spinner and no history is visible.

It's a big repo with a lot of commits and tags and branches.

The crash log:

```
Crash::: System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing
  its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (Failed to compare
  two elements in the array.)

----------------------------
Version: 8.36.0.0
OS: Microsoft Windows NT 10.0.19045.0
Framework: .NETCoreApp,Version=v8.0
Source: 
---------------------------



InnerException::: System.InvalidOperationException: Failed to compare two elements in the array.
   at System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource, Exception) + 0x38
   at System.Collections.Generic.ArraySortHelper`1.Sort(Span`1, Comparison`1) + 0x95
   at System.Collections.Generic.List`1.Sort(Comparison`1) + 0x75
   at SourceGit.ViewModels.BranchTreeNode.Builder.SortNodes(List`1 nodes) + 0x52
   at SourceGit.ViewModels.BranchTreeNode.Builder.SortNodes(List`1 nodes) + 0x65
   at SourceGit.ViewModels.BranchTreeNode.Builder.Run(List`1 branches, List`1 remotes, Boolean bForceExpanded) + 0x33f
   at SourceGit.ViewModels.Repository.BuildBranchTree(List`1 branches, List`1 remotes) + 0xbe
   at SourceGit.ViewModels.Repository.RefreshBranches() + 0x96
   at SourceGit.ViewModels.Repository.<RefreshAll>b__117_0() + 0x10
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread, ExecutionContext, ContextCallback, Object) + 0x4c
--- End of stack trace from previous location ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x20
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread, ExecutionContext, ContextCallback, Object) + 0xa3
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task&, Thread) + 0xcc

InnerException::: System.OverflowException: Value was either too large or too small for an Int32.
   at System.Number.ThrowOverflowException[TInteger]() + 0x2b
   at SourceGit.Models.NumericSort.Compare(String s1, String s2) + 0x31a
   at System.Collections.Generic.ArraySortHelper`1.PickPivotAndPartition(Span`1, Comparison`1) + 0x8b
   at System.Collections.Generic.ArraySortHelper`1.IntroSort(Span`1, Int32, Comparison`1) + 0xf5
   at System.Collections.Generic.ArraySortHelper`1.IntroSort(Span`1, Int32, Comparison`1) + 0x133
   at System.Collections.Generic.ArraySortHelper`1.IntroSort(Span`1, Int32, Comparison`1) + 0x133
   at System.Collections.Generic.ArraySortHelper`1.Sort(Span`1, Comparison`1) + 0x64
```

When I cloned again with `git clone ... --depth 300 --no-tags` then it works fine.

**Comment by love-linger** [comment]
It has been fixed by PR #612. You can download the latest CI build from Github Action.

**Comment by SlyEcho** [comment]
Yes, it is fixed in the latest build artifact. 👍
